livox_ros_driver/livox_ros_driver/lddc.cpp

```diff
@@ -518,6 +518,22 @@ uint32_t Lddc::PublishImuData(LidarDataQueue *queue, uint32_t packet_num,
   imu_data.linear_acceleration.y = imu->acc_y;
   imu_data.linear_acceleration.z = imu->acc_z;
 
+  // Fill in covariance matrices
+  cur_node_->param("angular_velocity_stddev", angular_velocity_stddev_, kDefaultAngularVelocityStdDev_);
+  cur_node_->param("linear_acceleration_stddev", linear_acceleration_stddev_, kDefaultLinearAccelerationStdDev_);
+
+  const double angular_velocity_covariance = angular_velocity_stddev_ * angular_velocity_stddev_;
+  const double linear_acceleration_covariance = linear_acceleration_stddev_ * linear_acceleration_stddev_;
+
+  imu_data.angular_velocity_covariance[0] = angular_velocity_covariance;

```

Comment:
I'm not quite sure why we're doing this. According to the IMU message documentation: 

    # If the covariance of the measurement is known, it should be filled in (if all you know is the 
    # variance of each measurement, e.g. from the datasheet, just put those along the diagonal)
    # A covariance matrix of all zeros will be interpreted as "covariance unknown", and to use the
    # data a covariance will have to be assumed or gotten from some other source
    #
    # If you have no estimate for one of the data elements (e.g. your IMU doesn't produce an orientation 
    # estimate), please set element 0 of the associated covariance matrix to -1
    # If you are interpreting this message, please check for a value of -1 in the first element of each 
    # covariance matrix, and disregard the associated estimate.

> A covariance matrix of all zeros will be interpreted as "covariance unknown"

It's perfectly valid to have an empty covariance matrix.

Reply:
Because the Kalman filter needs to have realistic covariance values to work properly. I should set the first element of the orientation covariance to -1 according to this text since the IMU does not provide orientation.

Reply:
I found a similar issue in a different repo: 
- https://github.com/ros-drivers/microstrain_mips/issues/19
- https://github.com/ros-drivers/microstrain_mips/pull/20

It looks like a similar fix, except that they populate the covariance matrix directly:
https://github.com/ros-drivers/microstrain_mips/blob/master/launch/microstrain.launch#L54

Is it more correct to populate it the way that you've done it?

Reply:
It's a valid point. In fact, it seems more clear to understand.
I already pushed the changes.